theano/gof/fg.py

```diff
@@ -478,8 +481,14 @@ def find_path_to(output_var, input_var):
 
         for node in new_nodes:
             assert node not in self.apply_nodes
-            self.__setup_node__(node)
-            self.apply_nodes.add(node)
+            prevent_addition = False
+            for n in self._removed_nodes :
+                if node is n :
+                    prevent_addition = True
```

Comment:
@nouiz I used ```is``` for comparing the exact instance of node object. AFAIK, ```in``` would do a value comparison and not the exact comparison.